docs/design/performance-testing.md

```diff
@@ -0,0 +1,62 @@
+# Performance Testing
+
+## Abstract
+
+This document proposes an initial environment to perform and measure automated benchmark results we can observe trends over time. 
+
+
+## Background
+
+We want to ensure the performance for uploads and downloads remains competitive and ideally world class. Our code base is changing rapidly and we’ve experienced degrading performance that we don’t have any visibility into what changed or when the performance dropped. 
+
+## Design
+
+### Goals
+* Keep it simple at first but give us something we can keep extending in the future. 
+* Measure both uploads and downloads and optionally replacements (causes a delete). 
+* Measure different workloads for both upload and download. 100 byte, 1MB, 10MB, 100MB and 1GB files. 
+* Measure each test 10 times and report the distribution graph of 50%, 75%, 90%, 99% percentiles. 
```

Comment:
https://medium.com/@egonelbre/benchmarking-with-details-1cf3e61d459d

Reply:
Yup I meant to reference this actually so I just added it to the document because there's already some great code that uses HdrHistogram format along with `hrtime` out there. I've linked it in the doc!

Reply:
Since we aren't measuring that many results or need to do it live, we can just keep all the results - no need to mess with HdrHistogram.

Reply:
The reason I mention HdrHistogram is because there's already easy to use graphs that follow the http://hdrhistogram.org format using go standard benchmarks. Also people in the industry are used to looking at graphs in this style. I use these graphs extensively in the past and being able to use a single format for latency distribution is extremely helpful. The code is already done for this pretty much.

Reply:
Ah, `hrtesting` gives you that as output as well.